spring-kafka/src/main/java/org/springframework/kafka/listener/ConsumerProperties.java

```diff
@@ -0,0 +1,325 @@
+	 */
+	private String groupId;
+
+	/**
+	 * Override the client id.
+	 */
+	private String clientId = "";
+
+	/**
+	 * A user defined {@link ConsumerRebalanceListener} implementation.
+	 */
+	private ConsumerRebalanceListener consumerRebalanceListener;
+
+	private Duration syncCommitTimeout;
+
+	/**
+	 * The commit callback; by default a simple logging callback is used to log
+	 * success at DEBUG level and failures at ERROR level.
+	 */
+	private OffsetCommitCallback commitCallback;
+
+	/**
+	 * Whether or not to call consumer.commitSync() or commitAsync() when the
+	 * container is responsible for commits. Default true.
+	 */
+	private boolean syncCommits = true;
+
+	private LogIfLevelEnabled.Level commitLogLevel = LogIfLevelEnabled.Level.DEBUG;
+
+	/**
+	 * Create properties for a container that will subscribe to the specified topics.
+	 * @param topics the topics.
+	 */
+	public ConsumerProperties(String... topics) {
+		Assert.notEmpty(topics, "An array of topics must be provided");
+		this.topics = topics.clone();
+		this.topicPattern = null;
+		this.topicPartitionsToAssign = null;
+	}
+
+	/**
+	 * Create properties for a container that will subscribe to topics matching the
+	 * specified pattern. The framework will create a container that subscribes to all
+	 * topics matching the specified pattern to get dynamically assigned partitions. The
+	 * pattern matching will be performed periodically against topics existing at the time
+	 * of check.
+	 * @param topicPattern the pattern.
+	 * @see org.apache.kafka.clients.CommonClientConfigs#METADATA_MAX_AGE_CONFIG
+	 */
+	public ConsumerProperties(Pattern topicPattern) {
+		this.topics = null;
+		this.topicPattern = topicPattern;
+		this.topicPartitionsToAssign = null;
+	}
+
+	/**
+	 * Create properties for a container that will assign itself the provided topic
+	 * partitions.
+	 * @param topicPartitions the topic partitions.
+	 * @deprecated in favor of {@link #ContainerProperties(TopicPartitionOffset...)}.
+	 */
+	@Deprecated
+	ConsumerProperties(org.springframework.kafka.support.TopicPartitionInitialOffset... topicPartitions) {
+		this.topics = null;
+		this.topicPattern = null;
+		Assert.notEmpty(topicPartitions, "An array of topicPartitions must be provided");
+		this.topicPartitionsToAssign = Arrays.stream(topicPartitions)
+				.map(org.springframework.kafka.support.TopicPartitionInitialOffset::toTPO)
+				.toArray(TopicPartitionOffset[]::new);
+	}
+
+	/**
+	 * Create properties for a container that will assign itself the provided topic
+	 * partitions.
+	 * @param topicPartitions the topic partitions.
+	 */
+	public ConsumerProperties(TopicPartitionOffset... topicPartitions) {
+		this.topics = null;
+		this.topicPattern = null;
+		Assert.notEmpty(topicPartitions, "An array of topicPartitions must be provided");
+		this.topicPartitionsToAssign = Arrays.copyOf(topicPartitions, topicPartitions.length);
+	}
+
+	@Nullable
+	public String[] getTopics() {
+		return this.topics; // NOSONAR
```

Comment:
Maybe `Arrays.copyOf()` as well? For consistency: see `getTopicPartitionsToAssign()`